network/p2p/handler.go

```diff
@@ -42,6 +44,20 @@ type Handler interface {
 	) ([]byte, error)
 }
 
+type NoOpHandler struct{}
+
+func (NoOpHandler) AppGossip(context.Context, ids.NodeID, []byte) error {
+	return nil
+}
+
+func (NoOpHandler) AppRequest(context.Context, ids.NodeID, time.Time, []byte) ([]byte, error) {
+	return nil, nil
+}
+
+func (NoOpHandler) CrossChainAppRequest(context.Context, ids.ID, time.Time, []byte) ([]byte, error) {
+	return nil, nil
+}
```

Comment:
Might be overthinking this but by returning `nil, nil` here this technically sends back an empty `AppResponse` message back to the requesting peer who will then drop the message because of an empty payload. We could alternatively error here with something like "not implemented" or something.